src/ServiceInsight/Framework/Modules/CoreModule.cs

```diff
@@ -21,6 +21,7 @@ protected override void Load(ContainerBuilder builder)
             builder.RegisterType<ServiceControlConnectionProvider>().InstancePerLifetimeScope();
             builder.RegisterType<DefaultServiceControl>().As<IServiceControl>().InstancePerLifetimeScope();
             builder.RegisterType<CommandLineArgParser>().SingleInstance().OnActivating(e => e.Instance.Parse());
+            builder.RegisterType<WorkNotifier>().As<IWorkNotifier>().SingleInstance();
```

Comment:
`WorkNotifier` takes an `IEventAggregator` in its constructor. The implementation of `IEventAggregator` is registered as `InstancePerLifetimeScope`. Does this not imply that `WorkNotifier` cannot be registered as `SingleInstance`?

Even if the *actual* behaviour of `IEventAggregator` is such that the container only ever creates one implementation, this is something that the code doesn't convey. I think it's better to register `WorkNotifier` as `InstancePerLifetimeScope` in this PR.

Then, if we want to change the implementations of both `IEventAggregator` and `IWorkNotifier` to `SingleInstance`, we can do that in a separate refactoring PR.